core/src/main/scala/org/apache/spark/sql/SnappyContext.scala

```diff
@@ -218,41 +219,125 @@ class SnappyContext protected[spark](
    * @todo provide lot more details and examples to explain creating and
    *       using sample tables with time series and otherwise
    * @param tableName the qualified name of the table
+   * @param samplingOptions
+    */
+  def createSampleTable(tableName: String, samplingOptions: Map[String, String]): DataFrame = {
+    val plan = createTable(catalog.newQualifiedTableName(tableName),
+      SnappyContext.SAMPLE_SOURCE, None, schemaDDL = None,
+      SaveMode.ErrorIfExists, samplingOptions,
+      onlyBuiltIn = true, onlyExternal = false)
+    DataFrame(self, plan)
+  }
+
+  /**
+   * Create a stratified sample table. Java friendly version.
+   * @todo provide lot more details and examples to explain creating and
+   *       using sample tables with time series and otherwise
+   * @param tableName the qualified name of the table
+   * @param samplingOptions
+   */
+  def createSampleTable(tableName: String
+      , samplingOptions: java.util.Map[String, String]): DataFrame = {
+    createSampleTable(tableName, samplingOptions.asScala.toMap)
+  }
+
+
+  /**
+   * Create a stratified sample table.
+   * @todo provide lot more details and examples to explain creating and
+   *       using sample tables with time series and otherwise
+   * @param tableName the qualified name of the table
    * @param schema
    * @param samplingOptions
-   * @param ifExists
    */
-  def createSampleTable(tableName: String, schema: Option[StructType],
-      samplingOptions: Map[String, String],
-      ifExists: Boolean = false): DataFrame = {
+  def createSampleTable(tableName: String, schema: StructType,
+      samplingOptions: Map[String, String]): DataFrame = {
     val plan = createTable(catalog.newQualifiedTableName(tableName),
-      SnappyContext.SAMPLE_SOURCE, schema, schemaDDL = None,
+      SnappyContext.SAMPLE_SOURCE, Some(schema), schemaDDL = None,
       SaveMode.ErrorIfExists, samplingOptions,
```

Comment:
Yes, First of all it is helpful for Java APIs.( Handling Option in Java is messy) .  Second I observed Spark's createExternalTable() also follows the same pattern. Perhaps they also faced the same issue with Java API. More importantly I feel its good to have two separate API , rather than forcing the user to pass None in case he does not want to pass a schema.